### PR TITLE
#58: Fix missing OS username

### LINKED ISSUE #58: clientOsUserName is not passed
When I connect to exasol via the go driver I can see that os_user is empty in `select * from exa_user_sessions;`

### PR DESCRIPTION
**Pull request by kaklakariada**
Closes #58

### CODE REVIEW

File: connection.go [hunk 1]
```diff
@@ -324,8 +325,10 @@ func (c *connection) login(ctx context.Context) error {
 		},
 	}
 
-	if osUser, err := user.Current(); err != nil {
+	if osUser, err := user.Current(); err == nil && osUser != nil {
 		authRequest.ClientOsUsername = osUser.Username
+	} else {
+		logCouldNotGetOsUser(err)
```
Comment (Nicoretti) [inline connection.go:331]: wouldn't you create an error and log it using the standard logging facitly? Also don't realy get why this is a fix, the error is just reported not realy dealt with.

File: errors.go [hunk 2]
```diff
@@ -140,6 +140,13 @@ func newFileNotFound(path string) DriverErr {
 		Parameter("path", path))
 }
 
+func logCouldNotGetOsUser(err error) {
```
Comment (Nicoretti) [inline errors.go:143]: wouldn't it make sense to move it up to the other log functions, so they are grouped?

Why not have a "generic" log function?

**Review by Nicoretti — APPROVED** [review]

File: errors.go (hunk 2 above)
Comment (kaklakariada) [reply errors.go:143]: Good point. The problem here is that error codes are numbered and as long as we don't have an error code crawler for Go that checks the uniqueness, it's easier to have all codes in one file.

Once we have the crawler, we can inline all these functions.

File: connection.go (hunk 1 above)
Comment (kaklakariada) [reply connection.go:331]: I am not sure what else to do in this case.

**Pull request merged**